src/keyutil/keyutil.c

```diff
@@ -277,11 +291,115 @@ static int verb_public(int argc, char *argv[], void *userdata) {
         return 0;
 }
 
+static int verb_pkcs1_to_pkcs7(int argc, char *argv[], void *userdata) {
+        _cleanup_(X509_freep) X509 *certificate = NULL;
+        _cleanup_free_ char *pkcs1 = NULL;
+        size_t pkcs1_len = 0;
+        int r;
+
+        if (!arg_certificate)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "--certificate= must be specified");
+
+        if (!arg_output)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "--output= must be specified");
+
+        if (argc < 2)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Input and output must be specified");
+
+        if (arg_certificate_source_type == OPENSSL_CERTIFICATE_SOURCE_FILE) {
+                r = parse_path_argument(arg_certificate, /*suppress_root=*/ false, &arg_certificate);
+                if (r < 0)
+                        return r;
+        }
+
+        r = openssl_load_x509_certificate(
+                        arg_certificate_source_type,
+                        arg_certificate_source,
+                        arg_certificate,
+                        &certificate);
+        if (r < 0)
+                return log_error_errno(r, "Failed to load X.509 certificate from %s: %m", arg_certificate);
+
+        r = read_full_file(argv[1], &pkcs1, &pkcs1_len);
+        if (r < 0)
+                return log_error_errno(r, "Failed to read PKCS1 file %s: %m", argv[1]);
+        if (pkcs1_len == 0)
+                return log_error_errno(SYNTHETIC_ERRNO(EIO), "PKCS1 file %s is empty", argv[1]);
+
+        /* Create PKCS7_SIGNER_INFO using X509 pubkey/digest NIDs */
+
+        _cleanup_(PKCS7_SIGNER_INFO_freep) PKCS7_SIGNER_INFO *signer_info = PKCS7_SIGNER_INFO_new();
+        if (!signer_info)
+                return log_oom();
+
+        if (!ASN1_INTEGER_set(signer_info->version, 1))
```

Comment:
These return ints and coding style is to not use boolean operators on ints, so please change all of these to compare with '== 0` instead.